runtime/bcutil/cfreader.c

```diff
@@ -2609,6 +2609,9 @@ checkClassVersion(J9CfrClassFile* classfile, U_8* segment, U_32 vmVersionShifted
 	}
 
 	buildError((J9CfrError *) segment, errorCode, CFR_ThrowUnsupportedClassVersionError, offset);
+	((J9CfrError *) segment)->errorMaxAllowedVersion = max_allowed_version;
+	((J9CfrError *) segment)->errorMajorVersion = majorVersion;
+	((J9CfrError *) segment)->errorMinorVersion = minorVersion;
```

Comment:
Line 2839 should have also checked that the provided `segment` is large enough to hold a `J9CfrError`: Please add it now.